substrate/frame/executive/src/lib.rs

```diff
@@ -308,11 +315,15 @@ where
 		let _guard = frame_support::StorageNoopGuard::default();
 		<AllPalletsWithSystem as frame_support::traits::TryState<
 			BlockNumberFor<System>,
-		>>::try_state(*header.number(), select)
+		>>::try_state(*header.number(), select.clone())
 		.map_err(|e| {
 			log::error!(target: LOG_TARGET, "failure: {:?}", e);
 			e
 		})?;
+		if select.any() {
```

Comment:
not a strong opinion, but I am a tad bit worried about the execution time of this. 

@liamaharon already brought up somewhere that running `try_state` is already time consuming af. Now, we're attaching something else that would potentially takes tens of minutes (I am guesing -- am I wrong?) to it. 

It is okay for now, but someday we might want to break what "try-runtime checks you run" into a configuration like this: 



```rust
/// What level of depth should each try-state check go into. 0 means be as fast as possible, 255 means be as slow. Possibly represent this with 3 level or so instead of 255. 
type TryStateLevel = u8;

enum TryStateLevelAlternative {
    Fast,
    Medium,
    Verbose
}

/// Which pallets should have their [`try_state`] involed in the context of [`TryStateChecks`]?
enum TryStatePallets {
    /// All pallets. 
    All,
    /// given number of pallets, round-robin based on block number. 
    RoundRobin(u8),
    /// given number of random pallets. Not sure if it is useful.
    Random(u8)
}


/// Final config, shoudl be passed into all `try_*` runtime APIs
struct TryStateChecks {
    decode_all: bool,
    try_state_pallets: TryStatePallets
    try_state_level: TryStateLevel 
}  
```

Reply:
@ggwpez mentioned it takes ~5s on Westend. So not that bad.

Reply:
On Rococo it was even faster. For now should be fine, but am also in favour of more granular settings.